test/langtools/jdk/javadoc/doclet/testMethodCommentAlgorithm/TestMethodCommentsAlgorithm.java

```diff
@@ -418,41 +387,67 @@ public interface T5 {
     }
 
     /*
-     * Takes a path to the java.base module, finds the Object.java file in
-     * there, creates a copy of that file _with the modified doc comment_
-     * for Object.equals in the provided destination directory and returns
-     * the path to that created copy.
+     * Locates source of the java.base module.
      */
-    private Path createPatchedJavaLangObject(Path src, Path dst, String newComment)
-            throws IOException {
-        if (!Files.isDirectory(src) || !Files.isDirectory(dst)) {
-            throw new IllegalArgumentException();
+    private Path findJavaBase() {
+        String testSrc = System.getProperty("test.src");
+        if (testSrc == null) {
+            // shouldn't happen
+            throw new SkippedException("test.src is not set");
         }
-        var obj = Path.of("java/lang/Object.java");
-        List<Path> files;
-        // ensure Object.java is found and unique
-        try (var s = Files.find(src, Integer.MAX_VALUE,
-                (p, attr) -> attr.isRegularFile() && p.endsWith(obj))) {
-            files = s.limit(2).toList(); // 2 is enough to deduce non-uniqueness
+        Path start;
+        try {
+            start = Path.of(testSrc).toAbsolutePath();
+        } catch (InvalidPathException | IOError e) {
+            throw new SkippedException("Couldn't make sense of '" + testSrc + "'", e);
```

Comment:
As a general stylistic convention, if you don't know much about the exception being caught or how the "rethrown" exception will be handled, I like to include the nested-exception message inr the rethrown message, as well as giving it in the cause.

So, for example,
```
throw new SkippedException("Couldn't make sense of '" + testSrc + "': " + e, e);
```

(note the `: " + e`)

Reply:
Firstly, this is not an API but a test. So we know how that exception is going to be handled. Secondly, I thought that when one translates an exception, one does at most one of these:

 - includes the exception's message in the rethrown exception message
 - sets the exception as a cause for the rethrown exception

That is, one does either of those, maybe none, but never both. Duplication makes an exception harder to read and defeats the purpose of chaining.

That said, I skimmed through the source and noticed that "including a message and chaining" is used; for whatever reason, in java.base it is used in the security area.